CoreLib/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+﻿# CoreLib
+set(CMAKE_CXX_STANDARD 20)
+
+set( CORELIB_HEADERS 
+	${CMAKE_CURRENT_SOURCE_DIR}/UniquePtr.h
+)
+
+set( CORELIB_SOURCES 
+	${CMAKE_CURRENT_SOURCE_DIR}/tests.cpp
+)
+
+set( ROOT ${CMAKE_CURRENT_SOURCE_DIR}/.. )
+
+set( GTEST_SOURCE_DIR ${ROOT}/External/googletest )
+
+add_executable(CoreLibTests ${CORELIB_SOURCES} ${CORELIB_HEADERS})
+
+target_link_libraries(CoreLibTests PRIVATE gtest gtest_main)
+target_include_directories(CoreLibTests PRIVATE ${GTEST_SOURCE_DIR}/include)
+
+if ( APPLE )
```

Comment:
We should think of how we can make CoreLib platform-agnostic later on.

Reply:
Well, it's actually platform-agnostic. But CMake just can't define custom defines (e.g. FUEGO_PLATFORM_MACOS or FUEGO_PLATFORM_WIN) in MSBuild project by itselfs. If we will build the project via CMake it can due to our top-level project CMakeFile (ROOT/CMakeLists.txt) defines macro-defines (FUEGO_PLATFORM_MACOS etc.) dependently on platform